libs/ui-watt/src/lib/components/button/watt-button.component.spec.ts

```diff
@@ -14,50 +14,90 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Type } from '@angular/core';
-import { render } from '@testing-library/angular';
-
-import { WattPrimaryButtonComponent } from './primary-button/watt-primary-button.component';
-import { WattSecondaryButtonComponent } from './secondary-button/watt-secondary-button.component';
-import { WattTextButtonComponent } from './text-button/watt-text-button.component';
-import { WattButtonVariant } from './watt-button-variant';
-import { WattButtonComponent } from './watt-button.component';
+import { render, screen } from '@testing-library/angular';
+import { WattIcon } from '../../foundations/icon';
+import {
+  WattButtonSize,
+  WattButtonTypes,
+  WattButtonVariant,
+} from './watt-button.component';
 import { WattButtonModule } from './watt-button.module';
 
-describe(WattButtonComponent.name, () => {
-  test.each([
-    ['text', WattTextButtonComponent],
-    ['primary', WattPrimaryButtonComponent],
-    ['secondary', WattSecondaryButtonComponent],
-  ] as readonly [WattButtonVariant, Type<unknown>][])(
-    'renders a %s button',
-    async (buttonVariant, buttonComponentType) => {
-      const view = await render(WattButtonComponent, {
-        componentProperties: {
-          variant: buttonVariant,
-        },
-        imports: [WattButtonModule],
-      });
-
-      const component = view.fixture.componentInstance;
-      expect(component.buttonComponentVariant).toBe(buttonComponentType);
-    }
-  );
+interface WattButtonAltOptions {
+  icon?: WattIcon;
+  loading?: boolean;
+  disabled?: boolean;
+  text?: string;
+  variant?: WattButtonVariant;
+  size?: WattButtonSize;
+}
+
+describe('WattButtonAltComponent', () => {
+  const renderComponent = async ({ ...options }: WattButtonAltOptions) => {
+    await render(
+      `<watt-button
+        variant=${options.variant}
+        size=${options.size}
+        icon=${options.icon}
+        [loading]=${options.loading}
+        [disabled]=${options.disabled}>
+          ${options.text ?? 'Text'}
+      </watt-button>`,
+      { imports: [WattButtonModule] }
+    );
+  };
+
+  it('shows text', async () => {
+    await renderComponent({ text: 'Text' });
+
+    expect(screen.getByRole('button')).toHaveTextContent('Text');
+  });
+
+  it('shows icon', async () => {
+    await renderComponent({ icon: 'plus' });
+
+    expect(screen.getByRole('img')).toHaveClass('mat-icon');
+  });
+
+  it('shows loading spinner', async () => {
+    await renderComponent({ loading: true });
+
+    expect(screen.getByRole('progressbar')).toHaveClass('mat-spinner');
```

Comment:
suggestion: change to 

```suggestion
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
```

to avoid testing for implementation details where possible.

Reply:
Fixed :)